test/extensions/filters/http/ext_authz/ext_authz_test.cc

```diff
@@ -830,6 +830,57 @@ TEST_F(HttpFilterTest, MetadataContext) {
             check_request.attributes().metadata_context().filter_metadata().count("hiphop.drums"));
 }
 
+// Test that filter can be disabled via the filter_enabled field.
+TEST_F(HttpFilterTest, FilterDisabled) {
+  initialize(R"EOF(
+  grpc_service:
+    envoy_grpc:
+      cluster_name: "ext_authz_server"
+  filter_enabled:
+    runtime_key: "http.ext_authz.enabled"
+    default_value:
+      numerator: 0
+      denominator: HUNDRED
+  )EOF");
+
+  ON_CALL(runtime_.snapshot_,
```

Comment:
maybe add prepareCheck here, too?

Reply:
Good idea. I was actually on the fence, because I usually hate copying and pasting boilerplate that is not needed.

However in this case, without `prepareCheck()` the test will segfault if a check is called. That isn't very nice experience, so I will add the call =).

Reply:
Actually... I can't add `prepareCheck()` as-is, because it will add an expectation that will only be satisfied if `check()` is called.... and this test verifies that `check()` is NOT called.